src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,495 @@
+          ##Calculate next value: y[n+1]. Equation (5.5) in Hairer et al, Vol 1
+          y[n+1] = y[n]
+          for j= 1:steporder
+            y[n+1] += h[n]*g[n,j]* Φstar[n,j]
+          end
+          #fixed step size code tested and gives correct result
+          ######################################################################
+          #Adaptive step selection
+          ######################################################################
+          #error bounds dictating if attempted step was successful
+          #See Hairer et al equations (7.3 - 7.4)
+
+          #Prediction of dy[n+1]
+          dy[n+1] = F(t[n+1],y[n+1])
+
+          #Don't use adaptive steps until n = order
+          if n<order
+            successful_step = true
+            h[n+1]=tspan[n]
+          else
+          #Once we have enough initial points, we use adaptive step selection
+          ##for this we need to calculate Φ[n+1,steporder+2]
+            for j = steporder+1 #k
+              if j== 1#0
+                b[n,j] = 1
+                Φ[n,j] = dy[n]
+                Φstar[n,j] = dy[n]
+              else j>1#0
+                b[n,j] = b[n,j-1]*(t[n+1]-t[n-(j-1)+1])/(t[n]-t[n-(j-1)])
+                Φ[n,j] =  Φ[n,j-1]- Φstar[n-1,j-1]
+                Φstar[n,j] = b[n,j]*Φ[n,j]
+              end
+            end
+            for j = 1:steporder
+              if j== 1
+                Φ[n+1,j] = dy[n+1]
+              else j>1
+                Φ[n+1,j] =  Φ[n+1,j-1]- Φstar[n,j-1]
+              end
+            end
+            Φ[n+1,steporder+2] = Φ[n+1,steporder+1] -Φstar[n,steporder+1]
+
+            ##Also we need to calculate g[n,steporder+1], g[n,steporder+2]
+            for j = 1:steporder+2#k+1
+              for q = 1:(steporder+2)-(j-1)#1:(k+1)-(j-1) #!!
+                if j ==0
+                  c[j,q]=1/q
+                elseif j==1
+                  c[j,q]=1/q/(q+1)
+                else
+                  c[j,q] = c[j-1,q] - c[j-1,q+1]*h[n]/(t[n+1]-t[n-(j-1)+1])
+                end
+              end
+              g[n,j] = c[j,1]
+            end
+
+            #From this we can calculate the error
+            @show LE = h[n]*(g[n,steporder+2] - g[n,steporder+1])*Φ[n+1,steporder+2]
+
+            if norm(LE,Inf)>1  || maxstep < abs(h[n])
+              successful_step = false
+              h[n] = h[n]/2
+            else
+              successful_step = true
+              @show h[n+1] = h[n]*(1/norm(LE,Inf))^(1/(steporder + 1)) #note: steporder = k + 1
+            end # if statement for successful step
+          end#while loop over successful step or not
+        end
+
+      #next step size selection
+      #@show norm(LE,Inf)
+     # @show h[n+1] = h[n]*(1/norm(LE,Inf))^(1/(steporder + 2)) #note: steporder = k + 1
+
+      #determine optimal next order
+
+      #move on to the next step
+      n= n+1
+    end #while loop over steps
+
+    return vcat(t), y
+end
+
+
+####################################
```

Comment:
I think you should forget about the implicit one for now and just debug the explicit one.